jsonp is called instead of json in cross domain requests

ubershmekel opened the pull request:
This was surprising to me and I'm still not sure how to handle cross-domain jsonp 404 responses.

arthurvr commented:
@ubershmekel This PR needs some updates. See comments above. It would be nice if you could do them?

ubershmekel commented:
Fixed the complaints and fixed the merge. From my testing `crossdomain` does not help detect the 404. The 404 being undetectable may be a security feature.

arthurvr commented:
Thanks @ubershmekel. What do we think @gnarf @kswedberg?

dmethvin commented:
I definitely like this formatting better. For this:

> Cross-domain requests may invoke "jsonp" unless explicitly "jsonp" is set to false.

I'd change it to something like:

> Cross-domain "json" requests are converted to "jsonp" unless the request includes `jsonp: false` in its request options.

arthurvr commented:
ping @ubershmekel

ubershmekel commented:
Sorry guys, I was abroad for a few weeks. It's a bit of a pain that every time I come back to this tiny change the pull request has merge conflicts.

I do like the proposed wording. If I fix it per @dmethvin would that be the final iteration?

arthurvr commented:
> It's a bit of a pain that every time I come back to this tiny change the pull request has merge conflicts.

For now feel free to not care about merge conflicts. I can take care of them while merging.

ubershmekel commented:
Merged with the new wording.

ubershmekel commented:
Ping @dmethvin @arthurvr before the next conflict.

dmethvin commented:
Whoa, sorry this got buried. LGTM!

arthurvr commented on entries/jQuery.ajax.xml:
```diff
@@ -74,8 +74,15 @@ $.ajax({
-          <ul><li>"xml": Returns a XML document that can be processed via jQuery.</li><li>"html": Returns HTML as plain text; included script tags are evaluated when inserted in the DOM.</li><li>"script": Evaluates the response as JavaScript and returns it as plain text. Disables caching by appending a query string parameter, "_=[TIMESTAMP]", to the URL unless the <code>cache</code> option is set ...
-          </li></ul>
+          <ul>
+            <li>"xml": Returns a XML document that can be processed via jQuery.</li>
```
> "xml"

I wonder if these shouldn't be in `<code>` elements. For the rest this LGTM!

arthurvr commented:
I'm going to make that trivial change myself and land this one. Thanks for your hard work, @ubershmekel , and sorry for it taking too long :)
